src/plugins/mer/mersettings.cpp

```diff
@@ -218,6 +221,181 @@ QMap<QString, MerEmulatorDeviceModel> MerSettings::deviceModels()
+
+    QStringList names = s_instance->m_deviceModels.keys();
+    std::sort(names.begin(), names.end(), [&collator](const QString &name1, const QString &name2) {
+        return collator.compare(name1, name2) < 0;
+    });
+
+    for (const QString &name : names) {
+        const MerEmulatorDeviceModel model = s_instance->m_deviceModels.value(name);
+
+        bool isModified = false;
+        if (!model.isSdkProvided()) {
+            if (!storedUserDeviceModels.contains(model.name())) {
+                isModified = true;
+            } else {
+                isModified = (model != storedUserDeviceModels.value(model.name()));
+            }
+        }
+
+        const QString label = QStringLiteral("%1 (%2x%3)%4")
+                .arg(name)
+                .arg(model.displayResolution().width())
+                .arg(model.displayResolution().height())
+                .arg(isModified ? QString(" *") : QString());
+        comboBox->addItem(label, name);
+        font.setItalic(!model.isSdkProvided());
+        const int i = comboBox->count() - 1;
+        comboBox->setItemData(i, font, Qt::FontRole);
+        comboBox->setItemData(i, model.isSdkProvided()
+                                               ? tr("SDK provided emulator profile")
+                                               : tr("User defined emulator profile"), Qt::ToolTipRole);
+    }
+
+    const int index = comboBox->findData(current);
+    comboBox->setCurrentIndex(index);
+}
+
+void MerSettings::deviceModelsReset()
+{
+    Q_ASSERT(s_instance);
+
+    QMap<QString, MerEmulatorDeviceModel>::iterator i = s_instance->m_deviceModels.begin();
+    while (i != s_instance->m_deviceModels.end()) {
+        if (i.value().isSdkProvided() == false)
+            i = s_instance->m_deviceModels.erase(i);
+        else
+            ++i;
+    }
+
+    s_instance->m_deviceModels.unite(deviceModelsRead());
+}
+
+QMap<QString, MerEmulatorDeviceModel> MerSettings::deviceModelsRead()
+{
+    QMap<QString, MerEmulatorDeviceModel> result;
+    const bool isSdkProvided = false;
+
+    // Reading user defined emulators
+    QSettings *settings = ICore::settings();
+    settings->beginGroup(QLatin1String(EMULATOR_SETTINGS_CATEGORY));
+    for (const QString &name : settings->childGroups()) {
+        settings->beginGroup(name);
+        const QSize resolution
+                = settings->value(QLatin1String(MER_DEVICE_MODEL_DISPLAY_RESOLUTION)).toSize();
+        const QSize size
+                = settings->value(QLatin1String(MER_DEVICE_MODEL_DISPLAY_SIZE)).toSize();
+        const QString dconf
+                = settings->value(QLatin1String(MER_DEVICE_MODEL_DCONF_DB)).toString();
+        settings->endGroup();
+
+        result.insert(name, MerEmulatorDeviceModel(name, resolution, size, dconf, isSdkProvided));
+    }
+    settings->endGroup();
+
+    return result;
+}
+
+void MerSettings::deviceModelsStore()
+{
+    Q_ASSERT(s_instance);
+
+    QSettings *settings = ICore::settings();
```

Comment:
I think we should use different settings file for this, much like it is done for e.g. MerSdk https://github.com/sailfishos/sailfish-qtcreator/blob/master/src/plugins/mer/mersdkmanager.cpp#L75

Reply:
Done. User defined profile now use file: `FileName MerSettings::deviceModelsFileName()`.